File: Rosalind/Ruby/gc.rb
=begin
Computing GC Content
http://rosalind.info/problems/gc/
=end

def gc_content(s)
  count = s.count('gc')
  count * 1.0 / s.length * 100
end

File.open(ARGV[0], 'r') do |f|
  count = 0
  f.each do |line|
    count += 1
  end
  puts count
  # max = ''
  # max_count = 0
  # until true
  #   name = f.gets.strip
  #   s = f.gets.strip
  #   count = gc_content(s)
  #   if count > max_count
  #     max_count = count
  #     max = name
  #   end
  # end
  # puts name
  # puts max_count
end

File: Rosalind/Javascript/gc.js
/*
Computing GC content
http://rosalind.info/problems/gc/
*/

function gcContent(s) {
  var count, i;
  count = 0;
  for (i=0; i<s.length; i++) {
    if (s.charAt(i) == 'G' || s.charAt(i) == 'C') {
      count += 1;
    }
  }
  return (count * 100 / s.length).toFixed(6);
}

//console.log(gcContent('CCACCCTCGTGGTATGGCTAGGCATTCAGGAACCGGAGAACGCTTCAGACCAGCCCGGACTGGGAACCTGCGGGCAGTAGGTGGAAT'));

var file = process.argv[2];
var fs = require('fs');
fs.readFile(file, {encoding: 'utf-8'}, function(err, data) {
  if (!err) {
    var lines = data.split('\n');
    var i = 0, hash = {}, current = "";
    while (i < lines.length) {
      if (lines[i].charAt(0) == '>') {
        current = lines[i].substr(1);
        hash[current] = '';
      } else {
        hash[current] += lines[i];
      }
      i += 1;
    }
    var name, item, highest = 0;
    for (item in hash) {
      if (gcContent(hash[item]) > highest) {
        highest = gcContent(hash[item]);
        name = item;
      }
    }
    console.log(name + '\n' + highest);
  } else {
    console.log('error');
  }
});
